test/e2e/test/elasticsearch/steps_mutation.go

```diff
@@ -75,19 +75,25 @@ func (b Builder) MutationTestSteps(k *test.K8sClient) test.StepList {
 	var clusterGenerationBeforeMutation, clusterObservedGenerationBeforeMutation int64
 	var continuousHealthChecks *ContinuousHealthCheck
 	var dataIntegrityCheck *DataIntegrityCheck
-	mutatedFrom := b.MutatedFrom
+	mutatedFrom := b.MutatedFrom //nolint:ifshort
 	isMutated := true
 	if mutatedFrom == nil {
 		// cluster mutates to itself (same spec)
 		mutatedFrom = &b
 		isMutated = false
 	}
 
-	masterChangeBudgetWatcher := NewMasterChangeBudgetWatcher(b.Elasticsearch)
-	changeBudgetWatcher := NewChangeBudgetWatcher(mutatedFrom.Elasticsearch.Spec, b.Elasticsearch)
+	watchers := []test.Watcher{
+		NewChangeBudgetWatcher(mutatedFrom.Elasticsearch.Spec, b.Elasticsearch),
+	}
+
+	isNonHAUpgrade := IsNonHAUpgrade(b)
+	if !isNonHAUpgrade {
+		watchers = append(watchers, NewMasterChangeBudgetWatcher(b.Elasticsearch))
+	}
```

Comment:
Could we also include the `NewChangeBudgetWatcher(...)` here to avoid the `WithChangeBudget(2, 2)` [previously added](https://github.com/elastic/cloud-on-k8s/commit/9d375cc128b52cef1e6d784b95b39e8b951e0344#diff-81988e08bfe062349de9bf20f8c0605cfffc3fd187250408ca118ee3e6a9956e) 
 to keep the watchers silent ?

```suggestion
	var watchers []test.Watcher
	isNonHAUpgrade := IsNonHAUpgrade(b)
	if !isNonHAUpgrade {
		watchers = []test.Watcher{
			NewChangeBudgetWatcher(mutatedFrom.Elasticsearch.Spec, b.Elasticsearch),
			NewMasterChangeBudgetWatcher(b.Elasticsearch),
		}
	}
```

Reply:
Good point I can change that.